Automattic/jetpack
# Google Analytics: extract API into the package

## Proposed changes:
* Move Google Analytics API code from Jetpack into the package.
* Add the `is_active` flag to indicate whether GA is active (currently module activation status is used).
* Modify the Status package to check activation status of an unavailable module to ease transition to settings when GA module gets removed from Jetpack.

### Other information:


## Jetpack product discussion
pfwV0U-4g-p2

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
1. Checkout the branch, connect Jetpack.
2. If reusing a site, deactivate the Google Analytics feature if enabled. Also please delete the `jetpack_wga` option from Jetpack site (`wp option delete jetpack_wga`) if it exists.
3. Go to "Jetpack -> Settings -> Traffic", purchase the plan.
4. Go there again and click on "Configure your Google Analytics settings".
5. In Calypso, scroll down to "Google" card and activate it.
6. Check `jetpack_wga`, confirm the setting `is_active: true` is now stored.
7. Go to the Jetpack site, find "Modules" section (`/wp-admin/admin.php?page=jetpack_modules`), and confirm that Google Analytics module is active.
8. Add a Measurement ID (e.g. `G-12345`), click "Save". Check `jetpack_wga`, confirm the Measurement ID and `is_active` are correct, and other settings show up and valid.
9. Reload the page, confirm Google Analytics is still active, and Measurement ID is preserved.
10. Load the site's frontend, check the page source code to confirm the Measurement ID is added to the page.
11. Deactivate "Add Google". Check `jetpack_wga` option, confirm `is_active: false`, and the rest of the settings didn't change.
12. Visit Jetpack's Modules page, confirm the module got disabled.
13. Check the frontend and confirm that the measurement ID has now disappeared.
14. Reload the Calypso page, confirm "Add Google" is still deactivated. Turn it on, confirm Measurement ID is still there.
